client/src/extension.ts

```diff
@@ -240,7 +240,10 @@ export function activate(context: ExtensionContext) {
   // language client, and because of that requires a full restart.
   context.subscriptions.push(
     workspace.onDidChangeConfiguration(({ affectsConfiguration }) => {
-      if (affectsConfiguration("rescript.settings.inlayHints")) {
+      if (
+        affectsConfiguration("rescript.settings.inlayHints") ||
+        affectsConfiguration("rescript.settings.codeLens")

```

Comment:
How do we know that `codeLens` config change requires a restart and other configurations don't?
Is there an easy rule one can follow to quickly figure that out? Or does one need to know the code of the entire extension plus the internals of the API for extensions, the spec?

Reply:
I added a clarifying comment.